### DISCUSSION TITLE
解决错误：go-chatgpt-api    | ERRO[0000] Failed to start, please try again later.

### DISCUSSION THREAD

**Discussion by tianlichunhong** [linweiyuan/go-chatgpt-api#258] [General]
经过反复尝试，终于知道为什么docker起不来。因为我是用VPS部署的
1、你的VPS已经被openai封杀了。这种情况，哪怕是curl -v https://chat.openai.com来测试都测试不出来。我测试发现一切正常。当我使用了代理ROXY=socks5://127.0.0.1:3708，就终于起来了。
2、起来后，依然没有出现下面这个语句：go-chatgpt-api    | INFO[0000] Service go-chatgpt-api is ready. 测试发现服务器也是不正常的。我之前配置使用的是    ports:  - 8080:8080。我改成了network_mode: host。服务就一切正常了。

下面是我用docker-compose启用的最终脚本。
version: "3" 
services:
  go-chatgpt-api:
    container_name: go-chatgpt-api
    image: linweiyuan/go-chatgpt-api
    network_mode: host
    environment:
      - PROXY=socks5://127.0.0.1:3708
    restart: unless-stopped

但是这个无法换端口，而且我也不想将端口对外暴露。请问我该如何解决？

**Comment by linweiyuan** [comment]
扯淡

1. Failed to start, please try again later. 是由于官网不稳定，这个时候你去官网使用也是不稳定的，这个过段时间再跑就好了

2. 和 network_mode 无关，因为你代理设置了 127.0.0.1，这样就会指向容器自己，而不是你的宿主机器，这个时候用 network_mode 当然可以了。另一个方法，socks5://你的服务器IP:3708，这个时候用 ports 则正常。

3. 国外服务器可以参考文档里 warp 的使用

**Comment by tianlichunhong** [comment]
Failed to start, please try again later的问题，我用代理立马起来了。如果不用代理，则一直起不来。我已经在不同的时段都尝试了。
我知道，如果使用socks5://127.0.0.1:3708这样的代理，容器指向127.0.0.1是自己的地址，需要network_mode: host才能实现指向主机的127.0.0.1。
我一直都在搜索如何不用network_mode: host，实现指向主机的127.0.0.1。使用
extra_hosts:
- "host.docker.internal:host-gateway"
socks5://host.docker.internal:3708
的方法依然解决不了。还是起不来。也就是依然没法指向主机的127.0.0.1。

**Reply by linweiyuan** [reply]
127.0.0.1 改为 172.17.0.1

**Reply by tianlichunhong** [reply]
我研究了一下。发现docker中，如果network_mode: bridge （默认）情况下，是无法访问宿主机127.0.0.1的。使用"host.docker.internal:host-gateway"或者使用172.17.0.1的方式，也仅仅访问宿主机的0.0.0.0的地址。容器和宿主机就相当两台独立的机器，一台机器是无法访问另外一台机器的127.0.0.1的地址的。能访问宿主机的127.0.0.1的方式，只能network_mode: host的网络模式。
所以我建议帮主，将这个容器设置可以通过配置文件自定义配置，该方案是比较成熟的。就是通过配置文件，可以改变容器内部接口的端口，等等。也为以后新增各种功能，实现自定义配置提供支持。
我已经不再使用本机127.0.0.1代理，目前go-chatgpt-api运行非常好！帮主的程序真是一级棒！！！

**Reply by linweiyuan** [reply]
看不懂
